dev/com.ibm.ws.logging/src/com/ibm/ws/logging/collector/CollectorJsonUtils1_1.java

```diff
@@ -268,16 +270,26 @@ private static String jsonifyTraceAndMessage(int maxFieldLength, String wlpUserD
             formattedValue.append(CollectorJsonHelpers.LINE_SEPARATOR).append(throwable);
         }
 
+        long logDataDatetime = logData.getDatetime();
+        String sequenceId = sequenceNumber.next(logDataDatetime);
+        logData.setSequence(sequenceId);
+
+        String messageIdVal = "";
+        if (message != null && eventType.equals(CollectorConstants.MESSAGES_LOG_EVENT_TYPE)) {
+            messageIdVal = CollectorJsonHelpers.parseMessageId(message);
+        }
+        logData.setMessageId(messageIdVal);
```

Comment:
should move the code for computing the messageId into the LogData.getMessageId method.